### PR TITLE
fix(graphql-connection-transformer): handle non-string-like sort keys

### PR DESCRIPTION
**Pull request by mikkom**
#### Description of changes

Fixes handling of non-string-like sort keys when using the `@connection(fields: [...])` directive. This was already handled correctly in the `makeGetItemConnectionResolver` method by using `$util.defaultIfNull` instead of `$util.defaultIfNullOrBlank` for non-string-like sort keys. In this change I have adapted the same logic to `makeGetItemConnectionWithKeyResolver` method and its call-site.

#### Issue #, if available

The issue has been discussed earlier in https://github.com/aws-amplify/amplify-cli/issues/2100#issuecomment-536628219

#### Description of how you validated changes

Includes a unit test for the non-string-like sort key case and check that the string-like sort key is also still handled correctly.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by mikkom** [comment]
@yuth Any chance you could take a look at this in the near future?

File: packages/graphql-connection-transformer/src/ModelConnectionTransformer.ts [hunk 1]
```diff
@@ -596,13 +596,44 @@ export class ModelConnectionTransformer extends Transformer {
...
+        if (
+          sortFieldType === STANDARD_SCALARS.Int ||
+          sortFieldType === STANDARD_SCALARS.Float ||
+          sortFieldType === STANDARD_SCALARS.Bolean
+        ) {
+          sortKeyIsStringLike = false;
+        }
+
+        sortKeyInfo = {
+          sortKeyIsStringLike,
```
Comment (cjihrig) [inline packages/graphql-connection-transformer/src/ModelConnectionTransformer.ts:625]: Can you perform the calculation inline here and get rid of the `sortKeyIsStringLike` variable completely:

```js
sortKeyIsStringLike: sortFieldType !== STANDARD_SCALARS.Int && ...
```

File: packages/graphql-connection-transformer/src/ModelConnectionTransformer.ts [hunk 2]
```diff
@@ -596,13 +596,44 @@ export class ModelConnectionTransformer extends Transformer {
...
+        const sortKeyField = keySchema[1] ? relatedType.fields.find(f => f.name.value === keySchema[1].AttributeName) : undefined;
+
+        // If a sort key field is found then add a simple sort key, else add a composite sort key.
+        let sortFieldType: string | undefined = undefined;
+        if (sortKeyField) {
+          sortFieldType = getBaseType(sortKeyField.type);
+        } else {
+          const compositeSortKeyType: SortKeyFieldInfoTypeName = 'Composite';
+          sortFieldType = compositeSortKeyType;
+        }
```
Comment (cjihrig) [inline packages/graphql-connection-transformer/src/ModelConnectionTransformer.ts:611]:
```suggestion
        const sortFieldType = sortKeyField ? getBaseType(sortKeyField.type) : 'Composite';
```

File: packages/graphql-connection-transformer/src/ModelConnectionTransformer.ts [hunk 3]
```diff
@@ -596,13 +596,44 @@ export class ModelConnectionTransformer extends Transformer {
...
+
+        let sortKeyIsStringLike = true;
+
+        // We cannot use $util.defaultIfNullOrBlank on non-string types
+        if (
+          sortFieldType === STANDARD_SCALARS.Int ||
+          sortFieldType === STANDARD_SCALARS.Float ||
+          sortFieldType === STANDARD_SCALARS.Bolean
```
Comment (cjihrig) [inline packages/graphql-connection-transformer/src/ModelConnectionTransformer.ts:619]: Based on this comparison, do we even need to worry about the case where `sortFieldType` equals `'Composite'`?

File: packages/graphql-connection-transformer/src/resources.ts [hunk 4]
```diff
@@ -284,10 +285,18 @@ export class ResourceFactory {
       ]);
     } else if (connectionAttributes[1]) {
       const sortKeyName = keySchema[1].AttributeName as string;
-      keyObj.attributes.push([
-        sortKeyName,
-        ref(`util.dynamodb.toDynamoDBJson($util.defaultIfNullOrBlank($ctx.source.${connectionAttributes[1]}, "${NONE_VALUE}"))`),
-      ]);
+      if (sortKeyInfo && !sortKeyInfo.sortKeyIsStringLike) {
```
Comment (cjihrig) [inline packages/graphql-connection-transformer/src/resources.ts:288]:
```suggestion
      if (sortKeyInfo?.sortKeyIsStringLike === false) {
```

File: packages/graphql-connection-transformer/src/__tests__/NewConnectionTransformer.test.ts [hunk 5]
```diff
@@ -352,6 +352,50 @@ test('Test ModelConnectionTransformer for One-to-One getItem case.', () => {
...
+    }
+    `;
+
+  const transformer = new GraphQLTransform({
+    transformers: [new DynamoDBModelTransformer(), new KeyTransformer(), new ModelConnectionTransformer()],
+  });
+
+  const out = transformer.transform(validSchema);
+  expect(out).toBeDefined();
+  console.log(out);
```
Comment (cjihrig) [inline packages/graphql-connection-transformer/src/__tests__/NewConnectionTransformer.test.ts:386]: Please remove this.

**Comment by cjihrig** [comment]
Closing due to lack of follow up. If you want to pick this back up, please let us know.

**Pull request closed without merging**